packages/website/docs/TextInput.md

```diff
@@ -1,20 +1,48 @@
-## Guidelines
+[Description]: # 
+[Text inputs allow to enter and edit short texts, such as names, emails or passwords.]: # 
 
-### Restrictions & Size
+### Best Practices
+Use text inputs when the user must enter one **short line of text.**
 
--   Users should be informed of any input restrictions, such as restrictions on character types or string length.
--   The text input area width should suggest the expected size of the user input. Try to forsee the text input length so as to avoid cutting it off. The text input area should be large enough to display the whole text input in most cases.
--   Avoid setting a character limit, or make sure to set it high enough to avoid constraining most user entries. If this is not possible, let users know how many characters are left.
--   Ensure that users with small screens can still enter all required information.
+The width of the text input should be proportional to the expected user input and large enough to display the entire text, in most cases. 
+The minimum width is 320px and can be incremented by multiples of 8.
 
-### Behaviour
+---
+### Labeling
+Keep labels short, under three words. 
+
+Use descriptive labels. Avoid action verbs. (E.g. Use "Email" instead of "Write your email")
+
+---
+### Help text and instructions
+Help text provide extra guidance on **what** information must be provided or **how** the user should fill it in (such as restricted characters).
+
+Help text should be **short one-liner**. 
+There are two ways to provide help text guidance: 
+ 
+- Written instructions : "Provide your business email." 
+- Example given: "E.g.: johndoe@acme.com" 
+
+A combination of both is also acceptable. 
+
+Text input should be self-explanatory as much as possible.
```

Comment:
Suggestion: 
The label of the text input should be self-explanatory as much as possible.